lib/system.nim

```diff
@@ -112,7 +112,7 @@ proc compileOption*(option, arg: string): bool {.
       discard "compiled with optimization for size and uses Boehm's GC"
 
 {.push warning[GcMem]: off, warning[Uninit]: off.}
-{.push hints: off.}
+# {.push hints: off.}
```

Comment:
I can revert the change i did here if needed, but what was the point of this? it prevents useful things, eg honoring `--processing:filenames`, or `--hint:cc` for modules (recursively) imported by system, which I want to see when compiling with those options.

Reply:
I've now reverted this diff and moved it in a dedicated PR, see https://github.com/nim-lang/Nim/pull/18034